src/sagas/repository.js

```diff
@@ -88,13 +89,35 @@ function* pollJobWorker(jobId, jobTypeSuccess, jobTypeFailure) {
  * Datasets.
  */
 
-export function* createDataset({ payload }) {
+export function* createDataset() {
+  const dataset = yield select(getCreateDataset);
+  const payload = {
```

Comment:
Let's call this `datasetRequest` (I'm getting confused with the multiple uses of `payload` below.)